source/common/upstream/health_discovery_service.cc

```diff
@@ -191,8 +191,10 @@ void HdsDelegate::processMessage(
 
       // add all endpoints for this locality group to the config
       for (const auto& endpoint : locality_endpoints.endpoints()) {
-        endpoints->add_lb_endpoints()->mutable_endpoint()->mutable_address()->MergeFrom(
-            endpoint.address());
+        auto* new_lb_endpoint = endpoints->add_lb_endpoints();
+        auto* new_endpoint = new_lb_endpoint->mutable_endpoint();
+
+        new_endpoint->mutable_address()->MergeFrom(endpoint.address());
```

Comment:
Is there a reason this part of the patch changed?

Reply:
No particular reason, thanks for pointing this one out. An artifact left over from some implementation attempts that I forgot to reverse. I reverted this change so it should be back as the original.